lang/java/avro/src/main/java/org/apache/avro/Conversions.java

```diff
@@ -117,6 +115,25 @@ public GenericFixed toFixed(BigDecimal value, Schema schema, LogicalType type) {
 
       return new GenericData.Fixed(schema, bytes);
     }
+
+    private static BigDecimal validate(LogicalTypes.Decimal decimal, BigDecimal value) {
+      int scale = decimal.getScale();
+      int valueScale = value.scale();
+      if (valueScale > scale) {
+        throw new AvroTypeException("Cannot encode decimal with scale " + valueScale + " as scale " + scale);
+      } else if (valueScale < scale) {
+        value = value.setScale(scale, ROUND_UNNECESSARY);
```

Comment:
this would be a safe operation, but wondering if there is any best-practice to set behind a flag that can be enabled/disabled?